backends/k8s/client.go

```diff
@@ -62,28 +65,36 @@ func NewK8sClient(kubeconfig string) (*Client, error) {
 		return nil, err
 	}
 
-	// Create the clientset
+	// Create the clientset (we use this for Node queries).
 	cs, err := kubernetes.NewForConfig(config)
 	if err != nil {
 		return nil, err
 	}
 	log.Debug(fmt.Sprintf("Created k8s clientSet: %+v", cs))
 
-	tprClient, err := buildTPRClient(config)
+	// Create the Calico backend client.  We use this to access all of the
+	// custom resources.
+	calicoK8sClient, err := calicok8s.NewKubeClient(&capi.KubeConfig{
+		Kubeconfig: kubeconfig,
+	})
+
 	if err != nil {
 		return nil, err
 	}
+
 	kubeClient := &Client{
-		clientSet: cs,
-		tprClient: tprClient,
+		clientSet:         cs,
+		calicoK8sClient:   calicoK8sClient,
+		nodeBgpPeerClient: resources.NewNodeBGPPeerClient(cs),
+		nodeBgpCfgClient:  resources.NewNodeBGPConfigClient(cs),
 	}
 
 	return kubeClient, nil
 }
 
 // GetValues takes the etcd like keys and route it to the appropriate k8s API endpoint.
 func (c *Client) GetValues(keys []string) (map[string]string, error) {
-	var kvps = make(map[string]string)
+	var vars = make(map[string]string)
```

Comment:
Yeah, let's not prematurely optimize.

We might want to change that if testing shows issues.